packages/vite/src/node/config.ts

```diff
@@ -497,6 +497,13 @@ export async function resolveConfig(
   const resolvedRoot = normalizePath(
     config.root ? path.resolve(config.root) : process.cwd(),
   )
+  if (resolvedRoot.includes('#')) {
+    logger.warn(
+      colors.yellow(
+        `There is a "#" in the path "${resolvedRoot}", which may hurt the performance or even break the build, please consider rename the folder name or remove the "#" in it.`,
```

Comment:
```suggestion
        `The project root contains the "#" character (${colors.dim(resolvedRoot)}), which may not work when running Vite. Consider renaming the directory to remove the "#".`,
```

Nit: Update the message a bit so it's clear that the `root` option is affecting it. I also removed the perf part as I think the bigger problem is it doesn't work well at first.